src/mon/FSCommands.cc

```diff
@@ -1202,6 +1202,118 @@ class RenameFilesystemHandler : public FileSystemCommandHandler
+	"credentials needs to be reauthorized to access the file systems "
+	"and its pools. Add --yes-i-really-mean-it if you are sure you wish "
+	"to continue.";
+      return -EPERM;
+    }
+
+    string fs1_name, fs2_name;
+    cmd_getval(cmdmap, "fs1_name", fs1_name);
+    cmd_getval(cmdmap, "fs2_name", fs2_name);
+    auto fs1 = fsmap.get_filesystem(fs1_name);
+    auto fs2 = fsmap.get_filesystem(fs2_name);
+
+    // Check that CephFSs for both gives names exists
+    if (fs1 == nullptr or fs2 == nullptr) {
+        if (fs1 == nullptr and fs2 != nullptr) {
+	  ss << "File system \"" << fs1_name << "\" doesn\'t exist on this "
+		"Ceph cluster.";
+	  return -ENOENT;
+	} else if (fs1 != nullptr and fs2 == nullptr) {
+	  ss << "File system \"" << fs2_name << "\" doesn\'t exist on this "
+		"Ceph cluster.";
+	  return -ENOENT;
+	} else {
+	  ss << "Neither file system \"" << fs1_name << "\" nor file "
+		"system \"" << fs2_name << "\" exists on this Ceph cluster.";
+	  return -ENOENT;
+	}
+    }
+
+    // Check that CephFS mirroring for both CephFSs is disabled.
+    if (fs1->mirror_info.mirrored or fs2->mirror_info.mirrored) {
+      if (fs1->mirror_info.mirrored) {
+	ss << "Mirroring is enabled on file system \""<< fs1_name <<
+	  "\". Disable mirroring on the file system after ensuring it's OK to "
+	  "do so, and then retry to swap.";
+	return -EPERM;
+      } else if (fs2->mirror_info.mirrored) {
+	ss << "Mirroring is enabled on file system \""<< fs2_name <<
+	"\". Disable mirroring on the file system after ensuring it's OK to "
+	"do so, and then retry to swap.";
+	return -EPERM;
+      } else {
+	ss << "Mirroring is enabled on file system1 \""<< fs1_name <<
+	      "\" as well as on \"" << fs2_name << "\". Disable mirroring on "
+	      "both the file systems after ensuring it's OK to do so, and then "
+	      "retry to swap.";
+	return -EPERM;
+      }
+    }
+
+    /*
+    if (!mon->osdmon()->is_writeable()) {
+      // not allowed to write yet, so retry when we can
+      mon->osdmon()->wait_for_writeable(op, new PaxosService::C_RetryMessage(mon->mdsmon(), op));
+      return -EAGAIN;
+    }
+    for (const auto p : fs->mds_map.get_data_pools()) {
+      mon->osdmon()->do_application_enable(p,
+					   pg_pool_t::APPLICATION_NAME_CEPHFS,
+					   "data", new_fs_name, true);
+    }
+
+    mon->osdmon()->do_application_enable(fs->mds_map.get_metadata_pool(),
+					 pg_pool_t::APPLICATION_NAME_CEPHFS,
+					 "metadata", new_fs_name, true);
+    mon->osdmon()->propose_pending();
+    */
+
+    string tmp_name = "fs774xxx";
+    auto f1 = [tmp_name](auto fs1) { fs1->mds_map.set_fs_name(tmp_name); };
+    fsmap.modify_filesystem(fs1->fscid, std::move(f1));
+    auto f2 = [fs1_name](auto fs2) { fs2->mds_map.set_fs_name(fs1_name); };
+    fsmap.modify_filesystem(fs2->fscid, std::move(f2));
+    auto f3 = [fs2_name](auto fs1) { fs1->mds_map.set_fs_name(fs2_name); };
+    fsmap.modify_filesystem(fs1->fscid, std::move(f1));
+
```

Comment:
Don't forget swappping the fscid as well. See

https://tracker.ceph.com/issues/58129#note-2

I suggest making that an option to turn on. It's confusing to do it by default.

Reply:
I've added an option that would swap FSCIDs too along FS names.

Reply:
Won't the user of this functionality expect that to be the default?